ArduCopter/mode_guided.cpp

```diff
@@ -548,17 +621,48 @@ void ModeGuided::posvel_control_run()
     uint32_t tnow = millis();
     if (tnow - posvel_update_time_ms > GUIDED_POSVEL_TIMEOUT_MS) {
         guided_vel_target_cms.zero();
+        guided_accel_target_cmss.zero();
         if (auto_yaw.mode() == AUTO_YAW_RATE) {
             auto_yaw.set_rate(0.0f);
         }
     }
 
-    // advance position target using velocity target
-    guided_pos_target_cm += guided_vel_target_cms * pos_control->get_dt();
+    bool do_avoid = false;
+#if AC_AVOID_ENABLED
+    // limit the velocity for obstacle/fence avoidance
+    // todo: this should use the position controller desired or target velocity
+    copter.avoid.adjust_velocity(guided_vel_target_cms, pos_control->get_pos_xy_p().kP(), pos_control->get_max_accel_xy_cmss(), pos_control->get_pos_z_p().kP(), pos_control->get_max_accel_z_cmss(), G_Dt);
+    do_avoid = copter.avoid.limits_active();
+#endif
 
     // send position and velocity targets to position controller
-    pos_control->input_pos_vel_accel_xy(guided_pos_target_cm, guided_vel_target_cms, Vector3f());
-    pos_control->input_pos_vel_accel_z(guided_pos_target_cm, guided_vel_target_cms, Vector3f());
+    if (do_avoid) {
+        guided_accel_target_cmss.zero();
+        pos_control->input_vel_accel_xy(guided_vel_target_cms, guided_accel_target_cmss);
+        pos_control->input_vel_accel_z(guided_vel_target_cms, guided_accel_target_cmss, false);
+        guided_pos_target_cm = pos_control->get_pos_target_cm();
```

Comment:
I think this lines means that the position target will be lost if avoidance becomes active.  unlike velocity control modes, the position only needs to be sent in by the external caller once.. there's no time out and so no need for the caller to repeatedly pass in the target position.

Reply:
Removed additional functionality